test/test_descriptor.js

```diff
@@ -34,3 +37,118 @@ describe('generate dp', () => {
     })
   })
 })
+
+describe('generate datapackage.json from local files: single file', () => {
+  it('check for csv data schema', function() { 
+    mock({
+      'data and Data and data and Data.csv': 'a,b,c\n1,1991-01-01,test1'
+    });
+    const expectedSchema = 
+    { 
+      name: 'scratchpad',
+      resources: [
+        {
+          path: 'data and Data and data and Data.csv',
+          name: 'data-and-data-and-data-and-data',
+          format: 'csv',
+          schema: {
+            fields: [
+              {
+                name: 'a',
+                title: '',
+                description: '',
+                type: 'integer',
+                format: 'default'
+              },
+              {
+                name: 'b',
+                title: '',
+                description: '',
+                type: 'date',
+                format: 'default'
+              },
+              {
+                name: 'c',
+                title: '',
+                description: '',
+                type: 'string',
+                format: 'default'
+              }
+            ]
+          }
+        }
+      ]
+    }
+    
+    const out = generateDescriptorForLocalFiles().then(val => {
+      assert.deepEqual(val, expectedSchema)
+      mock.restore();
```

Comment:
no, there is nothing about `done` in the readme 
https://github.com/tschaub/mock-fs